cmd/limactl/delete.go

```diff
@@ -44,6 +46,13 @@ func deleteAction(cmd *cobra.Command, args []string) error {
 		if err := deleteInstance(cmd.Context(), inst, force); err != nil {
 			return fmt.Errorf("failed to delete instance %q: %w", instName, err)
 		}
+		if runtime.GOOS != "windows" {
+			if err := autostart.DeleteStartAtLoginEntry(runtime.GOOS, instName); err != nil {
+				logrus.WithError(err)
+			} else {
+				logrus.Infof("The autostart file (%q) has been deleted", autostart.GetStartAtLoginEntryPath(runtime.GOOS, instName))
```

Comment:
I find it confusing that `limactl delete` will claim it deleted the autostart file even if it didn't exist to begin with.

I think `autostart.DeleteStartAtLoginEntry` should return `os.ErrNotExist` in that case, so you can skip any logging at all here.

Reply:
`DeleteStartAtLoginEntry` [returns `nil`](https://github.com/lima-vm/lima/pull/2151/files#diff-7c73aad62568bcacc05068a3c8cdf050206daaf75d7d1f2416c8faa19d385b32R62) if the file is absent.

Reply:
I know, that is the problem. When I delete the instance, I get

```
$ l delete default
INFO[0000] The qemu driver process seems already stopped
INFO[0000] The host agent process seems already stopped
INFO[0000] Removing /Users/jan/.lima/default under "*.pid *.sock *.tmp"
INFO[0000] Removing "/Users/jan/.lima/default/ha.sock"
INFO[0000] The autostart file ("/Users/jan/Library/LaunchAgents/io.lima-vm.autostart.default.plist") has been deleted
INFO[0000] Deleted "default" ("/Users/jan/.lima/default")
```

It claims to have deleted the autostart file, even though the file didn't exist. So the log entry is plain wrong.

That's why it needs to report an error back to the caller when the file doesn't exist, so the caller doesn't create a bogus log entry.

Reply:
Fair enough, I'll fix it, thanks 🙏

Reply:
The [`DeleteStartAtLoginEntry`](DeleteStartAtLoginEntry) now return an additional variable if file has been deleted. WDYT?